File: python/pages/loginPage.py
from basePage import BasePage
from selenium.webdriver.support.ui import Select, WebDriverWait
from selenium.common.exceptions import NoSuchElementException, TimeoutException

class LoginPage(BasePage):

    def __init__(self, driver):
        self.driver = driver
        self.driver.get('http://the-internet.herokuapp.com/login')
        self.assertTrue(self.driver.find_element_by_css_selector('#login'))

    def _element_present(self, how, what):
        try:
            self.driver.find_element(by=how, value=what)
        except NoSuchElementException, e: return False
        return True

    def _wait_for(self, locator):
        WebDriverWait(self.driver, 5).until(lambda driver : self._element_present('css', locator))

    def submit(self):
        self.driver.find_element_by_css_selector('#username').send_keys("username")
        self.driver.find_element_by_css_selector('#password').send_keys("password")
        self.driver.find_element_by_css_selector('#login').submit()
        self._wait_for('#flash')

File: ruby/pages/login_page.rb
require_relative 'base_page'

module Pages
  class Login < BasePage

    LOGIN_FORM            = '#login'
    INPUT_USERNAME        = '#username'
    INPUT_PASSWORD        = '#password'
    NOTIFICATION_MESSAGE  = '#flash'
    SUCCESS_MESSAGE       = '#flash.success'

    attr_writer :username, :password

    def initialize(driver)
      super
      visit('/login')
      find(LOGIN_FORM).should be_true
    end

    def now
      type(INPUT_USERNAME, username)
      type(INPUT_PASSWORD, password)
      submit(LOGIN_FORM)
      wait_for(NOTIFICATION_MESSAGE)
    end

    def success_message_present?
      element_present?(SUCCESS_MESSAGE)
    end

  end
end
